tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/errors/SqlErrorFactory.java

```diff
@@ -0,0 +1,136 @@
+package com.google.cloud.bigquery.utils.queryfixer.errors;
+
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.utils.queryfixer.entity.Position;
+import com.google.cloud.bigquery.utils.queryfixer.util.PatternMatcher;
+
+import java.util.List;
+
+/**
+ * A factory to create {@link BigQuerySqlError} from {@link BigQueryException}. Currently, it
+ * supports {@link TableNotFoundError}, {@link UnrecognizedColumnError}, and {@link
+ * FunctionNotFoundError}.
+ */
+public class SqlErrorFactory {
+
+  public static final String TableNotFoundRegex = "^Not found: Table (.*?) was not found";
+
+  public static final String UnrecognizedNameRegex =
+      "^Unrecognized name: (.*?)(; Did you mean (.*?)\\?)? at (.*?)$";
```

Comment:
If some keyword is not recognized, will it be "unrecognized name" too or it is only for column name?